diff -Naur A/core/arch/arm/plat-imx/conf.mk B/core/arch/arm/plat-imx/conf.mk
--- A/core/arch/arm/plat-imx/conf.mk	2020-09-02 11:06:09.222422790 +0530
+++ B/core/arch/arm/plat-imx/conf.mk	2020-09-07 16:20:06.852063956 +0530
@@ -72,6 +72,8 @@
 
 mx8qm-flavorlist = \
 	mx8qmmek \
+	mx8qm-iwg27s \
+	mx8qm-iwg27s-8gb \
 
 mx8qx-flavorlist = \
 	mx8qxpmek \
@@ -344,9 +346,9 @@
 $(call force,CFG_CORE_LARGE_PHYS_ADDR,y)
 endif
 
-ifneq (,$(filter $(PLATFORM_FLAVOR),mx8qxpmek mx8qmmek))
-CFG_DDR_SIZE ?= 0x80000000
-CFG_UART_BASE ?= UART0_BASE
+ifneq (,$(filter $(PLATFORM_FLAVOR),mx8qxpmek mx8qmmek mx8qm-iwg27s mx8qm-iwg27s-8gb))
+CFG_DDR_SIZE ?= 0x40000000
+CFG_UART_BASE ?= UART4_BASE
 endif
 
 ifneq (,$(filter $(PLATFORM_FLAVOR),mx8dxmek))
diff -Naur A/core/arch/arm/plat-imx/registers/imx8q.h B/core/arch/arm/plat-imx/registers/imx8q.h
--- A/core/arch/arm/plat-imx/registers/imx8q.h	2020-09-02 11:06:09.254422807 +0530
+++ B/core/arch/arm/plat-imx/registers/imx8q.h	2020-09-02 11:09:13.790553388 +0530
@@ -12,6 +12,7 @@
 #define UART1_BASE	   0x5a070000
 #define UART2_BASE	   0x5a080000
 #define UART3_BASE	   0x5a090000
+#define UART4_BASE	   0x5a0a0000
 #define SC_IPC_BASE_SECURE 0x5d1b0000
 #define CAAM_BASE	   0x31400000
 #define SECMEM_BASE	   0x31800000
